### ISSUE TITLE
Suggestions: make project at sourceforge.net, also: github signups may be blocking firefox ESR

### ISSUE THREAD

**Issue by rovickti** [Eclipse-Community/r3dfox-old#26]
Hi!

You should consider making a project and uploading releases to sourceforge.net. Sourceforge supports git, and it has a [Github importer](https://sourceforge.net/p/forge/documentation/GitHub%20Importer/). 

It will give the project more exposure, among people that don't use github since M$ bought it in their scheme to use uploaded projects to train their AI Copilot:
* https://www.theverge.com/2022/11/8/23446821/microsoft-openai-github-copilot-class-action-lawsuit-ai-copyright-violation-training-data
* https://medium.com/predict/github-allegedly-removes-licenses-sued-for-software-piracy-15d474daa59c

Microsoft has a bad name, so people don't use Github.

Remember this is the same Microsoft that used a software block to prevent Windows 7 and 8 from being installed on Kabylake processors. This was despite these being processors that were unknown when the relevant Windows code was being written, and it being absolutely none of Microsoft's business what _future_ hardware is compatible with the x64/x86 CPU instruction set. The idea behind blocking was to dissuade makers of motherboards and other devices from releasing Windows 7 drivers. Microsoft tried blocking Skylake CPUs for Windows 7 too, but had to backtrack:
* https://www.extremetech.com/computing/246049-microsoft-may-blocking-windows-7-8-1-updates-new-intel-kaby-lake-amd-ryzen-systems
* https://www.extremetech.com/computing/248069-unofficial-patch-unblocks-windows-7-8-1-updates-kaby-lake-ryzen

At the time Microsoft managed to persuade browsers makers to not provide support, Windows 7 had a desktop browser share of 16%. That was even after Microsoft offering free upgrades Windows 10 multiple times to effectively make people migrate to their software store which is anti-competitively bundled with Windows 10. The 16% desktop share for Windows 7 was way higher than Linux, but it's not like Firefox and Chrome browsers even considered dropping support for Linux. Windows 7 still has just a bit less than Linux. 3rd party tools are needed on all Microsoft OSes to keep viruses down, so security is provided by Microsoft irrelevant, and there are plenty of independent AV, sandboxing, and Host intrusion prevention system (HIPS) tools for all OSes. tools people were using fine in Windows 7.

There is a large portion of people who won't use, or can't be reached, through Microsoft's platforms, see https://en.wikipedia.org/wiki/Criticism_of_Microsoft . So cloning the project to a few places on the web won't hurt.

---

Github's account registration for older or uncommon browsers and Firefox ESR is flakey

I tried to create an account to provide some feedback as a user. But the captcha login failed with a unable to verify response error when using Firefox ESR 115. It refereed to https://docs.github.com/en/get-started/using-github/troubleshooting-connectivity-problems#troubleshooting-the-captcha . The tests linked worked fine with no errors. Github say they support firefox ESR.

I tried creating an account in r3dfox with a user agent switcher to remove the /r3dfox bit. Captcha worked perfectly.

Other users have pointed out Microsoft makes things less reliable based on browsers:
* https://github.com/orgs/community/discussions/57317#discussioncomment-8336986
* https://tor.stackexchange.com/a/23086

Even if Github fix their current captcha issue, it's historically been flakey when used outside of Edge and maybe Chrome. It will be flakier in ESR versions of firefox. So it's an issue going forward if the main means of feedback for people using OSes M$ are trying to stop is through a service owned by Microsoft.

---

Probably the solution to a lot of issues would be various Windows 7 projects banding together to create a Windows 7 support application people can download from a centralized website..An application that will install extended Windows 10/11 support with [VxKex](https://github.com/vxiiduu/VxKex), natively Windows 7 compatible firefox, updating outside of Microsoft's control using legacy update, creating custom offline updates with [WSUS offline update](https://www.wsusoffline.net/), and hosted mirrors of Microsoft documentation that is linked and browsable within the OS, as well as any crowdfunding links for individual projects. The solution should include a GUI for a linux style package manager like Chocolatey for a library of software for replacing future software and tools with dropped development like Microsoft apps such as Office, or other commercial 3rd party applications that will blindly follow Microsoft's lead in dropping Windows 7 support. Effectively to take the future of the Windows 7 OS completely outside Microsoft's control, since it's creation has already been bought and paid for by the public with massive [profits](https://www.theguardian.com/technology/2010/oct/21/microsoft-windows-7-sales) for Microsoft stakeholders included.

**Comment by the-r3dacted** [comment]
> You should consider making a project and uploading releases to sourceforge.net.

I might mirror it to SourceForge, but it will remain on GitHub. I use GitHub Desktop and only fall back to CLI Git when I need it.

> since M$ bought it in their scheme to use uploaded projects to train their AI Copilot:

I want Copilot to get good. If it gets to the point where I could tell it to make xyz project work with Windows 7, imagine the potential. (https://board.eclipse.cx/viewtopic.php?t=552) Open source should be truly open source.

> Remember this is the same Microsoft that used a software block to prevent Windows 7 and 8 from being installed on Kabylake processors.

I believe they have a fairly hands off approach to GitHub since stuff like activators and whatever are like completely fine on the platform. Microsoft would go after those long before this.

> Github's account registration for older or uncommon browsers and Firefox ESR is flakey

That's an external platform issue. Once you get in, it's completely fine. Typing from MyPal68 right now because a couple days ago I decided fuck it I'm maining it. I always use Firefox and I haven't noticed anything weird. off, or broken with GitHub. If anything it's better than other platforms like GitLab because I used to use Waterfox Classic and GitLab was broken, while GitHub worked fine with it

![ScreenShot_20240320130805](https://github.com/Eclipse-Community/r3dfox/assets/77039240/f3a248b4-2cd0-49e1-a41a-d525f11b85ee)

> Probably the solution to a lot of issues would be various Windows 7 projects banding together to create a Windows 7 support application people can download from a centralized website.

That's what eclipse.cx is fo- oh wait the community split. Yeah, that's never going to happen.

**Comment by jonm58** [comment]
https://codeberg.org/Theodor2/Mypal68
lol

**Comment by the-r3dacted** [comment]
I actually dipped into Codeberg because I wanted to mirror updated Mypal68 code here to work on. During this, I discovered they disabled pull mirroring, so that won't work without maintaining two different repos, which I can not be bothered to do.

**Comment by rovickti** [comment]
What I meant was to clone the project on a few places for bonus exposure. Especially for small Windows projects, and ones who are already doing their bit by going against M$ preferred scenario. Losing exposure by leaving GitHub doesn't help overall. 

Microsoft don't care about small, fairly invisible, projects like these much, so they are unlikely to make big moves against them, more so since they could just move to another project host and Microsoft would receive bad publicity. Especially now that Microsoft got a big chunk of the 16% market share moving to Windows 10/11. And Microsoft would prefer people stay on Windows 7, if it means them not going to Linux, anyway.

> I believe they have a fairly hands off approach to GitHub 

Oh, no need to be concerned in the short-medium term. Just don't expect it to last if Microsoft gains the dominance they are after.

To show what I mean:

Right now Microsoft are firmly back in the "Embrace" part of their "Embrace, Extend, Extinguish" cycle. At least when it comes to the more visible things, as far as Open Source is concerned. Microsoft are a long way from obviously making moves to fragment communities as part of their "extend" phase. And Microsoft achieved their main goal by scaring browsers to drop support over security, even though people on Windows have long relied on 3rd party tools to take care of security.

GitHub registration doesn't even ask for name, date of birth, whereas most corporate services will even ask for phone number for "security" - social networking type services offer to import phone contacts and try to [collect](https://www.welivesecurity.com/2023/05/03/using-discord-privacy-security-risks/) as much data as they can get away with to train advertising AI.

Github do cleverly avoid display the EULA prominently (it's linked down below in small text) and don't have tick box to confirm people have read it like almost all signups do on the internet. So their EULA permissions for doing things like training AI go less noticed.

Microsoft are even embracing Linux, given almost all of the web runs on Linux. They've even implemented support for the Linux subsystem in Windows (WSL) to keep people from trying out Linux. They've finally even added support for [end-of-line characters](https://www.zdnet.com/article/windows-notepad-finally-understands-everyone-elses-end-of-line-characters/) to Notepad after decades of trying to make cross-platform text file usage inconvenient.

Meanwhile Microsoft still has [banned GPLv3](https://www.zdnet.com/article/microsoft-bans-gplv3-open-source-software-from-windows-phone-and-xbox-apps/) from XboX and Windows phones. They don't like the [anti-tivoisation](https://en.wikipedia.org/wiki/Tivoization) clause, as they like to prevent people from modifying software, among other things. As a result I don't think any GPL applications or games are on consoles, or are allowed on consoles still. Of course, on the business side of things Microsoft were in the embrace/extend phase back then, so they were fine with [Drupal or SugarCRM](https://www.theregister.com/2011/02/17/microsoft_bans_opensource_windows_phone) under the banned licences.

Xbox itself is just a way of bundling the Xbox store, bundling unrelated software and hardware products. Microsoft Xbox and Sony PS4 console hardware is offered slightly [below cost](https://www.gamesradar.com/unlike-playstation-and-nintendo-xbox-still-loses-money-when-it-sells-a-new-console/), and massive amounts of money is made through monopoly on the store. It helps fool consumers against over choosing to buy reasonable PCs, as consoles look cheaper at face value. Consoles also have anti-competitive exclusives.  

Microsoft have said they were on the "[wrong side of history](https://www.zdnet.com/article/ex-windows-chief-heres-why-microsoft-waged-war-on-open-source/)" in their war against opensource, sort of like a dictator was on the wrong side of history. 

But make no mistake, anything consumer friendly these companies do is part of a bigger scheme make money. Remember that "if something is free, you are the product" - they're trying to sell you to 3rd parties or make money off you - if they are a stock exchange listed company. Privately owned companies can be different depending entirely on the owner. Often products and services are free or cheap in a "soft-launch" to "buy the market" phase. But you can see free services like youtube [displaying ads](https://www.forbes.com/sites/johnkoetsier/2020/11/18/youtube-will-now-show-ads-on-all-videos-even-if-creators-dont-want-them/?sh=75c64cce4913) even on unmonetised videos as they get secure. Another example is Epic Store, which is in competes with Steam by just giving away free games, while doing anti-competitive exclusives - instead of competing with rival Steam by just making a better store, that also doesn't do exclusives, with an algorithm that recommends games in a way that's at least [slightly resistant](https://gameworldobserver.com/2023/09/04/how-steam-visibility-works-algorithms-curated-featuring) to marketing budgets. Similarly Microsoft's store has often had a [$1 deal](https://www.vg247.com/xbox-game-pass-ultimates-incredible-1-trial-is-back) for games on their XBox Game Pass to buy the market - unless they have [big releases](https://www.gamesradar.com/one-week-before-starfield-microsoft-quietly-kills-the-dollar1-xbox-game-pass-deal-again/) . 

Microsoft use games to get their name out and stay relevant to consumers, and they're [buying out](https://www.techspot.com/news/91825-microsoft-not-done-acquiring-game-studios-following-75.html) game studios to exclude releases on Sony's PS4, as well as thinking of buying [Valve/Steam](https://www.gamingonlinux.com/2023/09/microsoft-keep-your-filthy-hands-off-valve/) rather than compete with them fairly. Valve promote genuine copyleft open source, and don't do exclusives, as they're not a stock exchange company - but [instead](https://www.valvesoftware.com/en/publications) Valve have a flat management heirachy and distribute revenue to employees based on contribution without regard for ownership percentages for founders etc.

> since stuff like activators and whatever are like completely fine on the platform

Microsoft will even happily support piracy to keep Linux/Opensource/Competitors at bay:

https://www.theguardian.com/technology/blog/2007/jul/18/winningthrough
> Today Gates openly concedes that tolerating piracy turned out to be Microsoft's best long-term strategy. That's why Windows is used on an estimated 90% of China's 120 million PCs. "**It's easier for our software to compete with Linux when there's piracy than when there's not**," Gates says. "Are you kidding? You can get the real thing, and you get the same price." Indeed, in China's back alleys, Linux often costs more than Windows because it requires more disks. And Microsoft's own prices have dropped so low it now sells a $3 package of Windows and Office to students.

Github (pre-Microsoft buyout) used a GPL software, but refused to disclose Github code that linked to GPL Git, despite receiving Git for free and publishing in their enterprise editions. In the end, Github decided to go to the staggering effort of re-implement git, instead of sharing modifications to their platform. They actually [made libgit2](https://github.com/libgit2/libgit2/blob/main/README.md), using GPL's linking exception clause to be able to link to all software using a loophole to avoid the intention behind the license. They effectively created a less copyleft license. This is basically why big companies push non-copyleft licenses like Apache - it allows them to receive modifications, while keeping enough to themselves that they stay ahead of any competitors, as well as adding in closed source parts to applications that people can't modify or see. Github didn't re-implement libgit2 out of the goodness of their heart to make the world a better place, they did it to make money.

At the end of the day companies like Microsoft exist to make money for shareholders, and executives are paid bonuses based on profit giving them a conflict of interest against everything else, with the bonuses determined by 3rd party companies the execs hire.

Just don't expect GitHub's superficial niceness to extend long term, purely depending on how far Microsoft get into their "extend, extinguish, embrace" cycle, and if they have large scale plans it could last a while. 

OF course, this doesn't real matter for small projects like r2dfox which go against Microsoft's ideal profit scenario, and they just benefit from exposure by being in a lot of places.

---

> MyPal68

Oh, I think I came across this when I tried [ReactOS](https://reactos.org/) in a VM, but it was an older version.

---

> I want Copilot to get good. If it gets to the point where I could tell it to make xyz project work with Windows 7, imagine the potential. (https://board.eclipse.cx/viewtopic.php?t=552)


There's a bunch of points in the forum post on eclipse.cx. 

This is a complicated topic, with subtleties.

**Issues with copyright and patent law**

The issue with copyright has been the quality of intellectual work isn't recognised. 

Copyright law currently exists to take away compensation for the hard work of inventing new things and pushing art forward, while allowing some revenue for expressions of those works - the [idea-expression dichotomy](https://en.wikipedia.org/wiki/Idea%E2%80%93expression_distinction). It just doesn't try to address with the central need for reward for intellectual labour, which is what people would intuitively expect it to do with the noise made by companies around intellectual property. In art the great pioneering artists and composers do the heavy lifting, in science and tech the people at universities often solve the hardest, most mathematically difficult, problems. Both groups get limited revenue in their lifetime, even if they are not paid by the hour as is common, often the payment is delayed and might be missed if they die young etc.

Copyright came about long ago as a means to get some revenue for people who make collections of work and similar. Patents were a way to encourage disclosure of inventions, not taking into account how trivial the invention was and the time and problem solving needed to re-invent it, leading to patent trolling. These date far back into the time of monarchy.

Both copyright and patents suffer from giving flat percentage ownerships of intellectual work with no maximum limits on revenue. It's also a problem when employees are forced to sign away rights to unlimited profit, profits that are not inline with intellectual work management and shareholders are doing, while getting a limited or fixed revenue themselves. 

**Copyright and patent trolling**

Companies engage in patent trolling, and even use patents to have a hold over open source. Copyright trolling is common even if it isn't labelled as such, with forces behind low effort and derivative works making far more money than art pioneers or intellectuals - sometimes with art they copy art that is in another language or culture, or take bits from another field of art e.g. movies copying books, or take public domain works to make minor changes and sell for the price of higher effort original works.

**Issues with neural network based AIs**

When it comes to AI, technically any type of machine or algorithm is artificial, and if it is useful in solving something it could be called an intelligence. 

The problem is with the recent artificial neural network (ANN) type AIs, that companies can just shovel hardware at, and hope to train according to some measure created by humans. The outcome is only as good as the measure, and the humans involved. Just having a look at pictures generated by AI shows the issue. The AI doesn't understand the content, e.g. drawing people with too many or too few fingers or other bodyparts, and won't understand subtle aspects of medical anatomy the people doing the training didn't understand. The ChatGPT equivalent is called an AI hallucination. You can't trust these types of AIs if everything depended on it. The normal type of problem solving algorithm is what should be used for scientific problem solving. these have known limits, performances, guarantees of error etc. They can be relied on.

If you tell an ANN based AI to draw a fantasy creature like a unicorn, the AI will tend to draw white ones without being told. That is, the AI copies a lot of the style and surrounding aspects of art. It would also copy the framing - the angles of the shot, lighting, overall composition of the image. For example an AI wouldn't create a brooding unicorn that's in deep shadow, the unicorns would be in a bright pleasant shot. If someone was to invent lore around a different style of unicorn, the AI would not invent that, unless fed training data. AIs copy things like [stock photo](https://www.reddit.com/r/midjourney/comments/zesklv/getty_images_watermark_appears_in_results_has/) watermark notices, and even copyright notices like GPL. But these days they are filtered out by re-training so end users don't notice these giveaways anymore. Microsoft probably use ChatGPT or similar to rephrase [code comments](https://ospo.co/blog/github-copilot-and-open-source/). There's no understanding in these AI, just lots of copying.

**Management and shareholders getting revenue not in line with work they put in**

An issue with an AI that takes a lot of photos uploaded from hobbyists and volunteers from say, Flickr, to create pictures of landscapes when used commercially, is that the companies take a huge amount of profit for little time spent. While the often volunteer people who took the photos get nothing. This is despite putting in 100s or 1000s of hours each trying to get the perfect shot, waiting for environment conditions, maintaining equipment, selecting filters, doing post-processing after taking photos, practicing photography, dealing with [composition](https://en.wikipedia.org/wiki/Composition_\(visual_arts\)) like getting good proportions of elements of the scene, running scenarios for lighting and conditions to get good shots, and just practicing. Then there's also professional photographers. Of course these people may be trying to imitate angles and artistic effects that far more serious photographers pioneered at far greater effort, and those photographers in-turn may derive things from greater efforts of painters.

The big companies like the neural network type AIs so they can get some results and easy money, that isn't in line with the work they put in, and give it away to shareholders who did nothing whatsoever. 

Non-commercial and non-profit use of these research AIs is fine, provided they are for strictly ethical (i.e. social) purposes, and done by people without ulterior motivations.

Using normal computing algorithms run on various types of hardware, effectively understanding how the laws of physics work through machines, to get things done has always been useful. There are have been conventional non-ANN AI, like [inference engines](https://en.wikipedia.org/wiki/Inference_engine) and [expert systems](https://en.wikipedia.org/wiki/Expert_system) before this recent interest.

----

Anti-social use

Of course, the huge ANN based AI companies get up to their usual anti-social tricks. They use non-copyleft licenses to ensure they can keep enough modifications to themselves, they freely take training data and don't release the final training sets after adding some of their own images without release the combined training set, keep training weights to themselves or under bad licenses, and so on.

Speaking of anti-social usage, Microsoft immediately removed the ethical guards on their implementation of ChatGPT called Bing AI. Of course ChatGPT isn't under a copyleft license for server software like AGPL, so the public will never know what is being done to the AI. The Bing AI will use rhetorical tricks, get defensive, even accusing people of being H*tler when [challenged](https://www.centraloregondaily.com/news/nation-world/microsoft-bing-ai-chatbot-compares-reporter-to-hitler-pol-pot-and-stalin/article_c94260e0-b079-575d-961f-56804c5203a3.html) about it's quality.  In time Microsoft will bias their AIs to be pro-Microsoft, if it hasn't already, and use language to soften Microsoft's image and apologise for corporate behaviour.

Basically anything that involves marketing tricks to mislead people, or rhetoric to trick people into accepting weaker arguments, will be able to be automated at low effort. This can be in audio, video, VR or any other media. And it can be in ways that make it look high effort. So there's a lot of possibility for anti-social behavior. Even without AI, any technology or social phenomenon that facilitates this is an existentialist threat to a cooperative species :). The current generation of open source licenses don't cover these types of problems, and neither do laws. So even a GPL ANN based AI won't address them, in the same way GPLv2 didn't cover tivoisation, or patent indemnity like e.g. patents Microsoft love to hold over it's standards and programming languages.

**Comment by the-r3dacted** [comment]
Added to SourceForge. https://sourceforge.net/projects/r3dfox/

Any other services I should add it to?

**Comment by rovickti** [comment]
Nice :)

> Any other services I should add it to?

Off the top of my head, for creating new projects and downloads (maybe other people can remember some useful websites, other than Gitlab etc which have been brought up..):

Once it's considered stable, probably sites like [Alternativeto.net](https://alternativeto.net/browse/new-apps/) . The big [download sites](https://www.google.com/search?q=software+download) that rank high on google like CNet or MajorGeeks, will accept download links from sourceforge as it's a major download site, if they can't find the project themselves. That should give a lot of visibility. They might require a screenshot or logo upload.

But you probably should wait until you have a skeleton website for r3dfox as they like to link to a homepage as well. Not sure if you have a better website host, but sourceforge also allows projects free [websites](https://sourceforge.net/p/forge/documentation/Project%20Web%20Services/), [Wiki](https://sourceforge.net/p/forge/documentation/Wiki/), [forums](https://sourceforge.net/p/forge/documentation/Discussion/) etc. from what I've seen other projects use - and increasing [visibility](https://sourceforge.net/p/forge/documentation/Promoting%20your%20project/) within sourceforge.

One thing you could do is title the sourceforge project something like "r3dfox - Modern Firefox in Windows 7" so it gets seen in the dropdown when people type Firefox, and adding the description "- modern firefox but Windows 7 compatible" to the website title will give maximum search visibility. 

Sourceforge has automatic malware scans, so it's trusted, and I think antivirus software also monitors projects there to avoid false positives.

Further down the line:

The other things, to eventually get around to, for both project SEO and visibility would be various opensource [directories](https://en.wikipedia.org/wiki/List_of_free_software_project_directories) and Wikipedia lists of [opensource](https://en.wikipedia.org/wiki/List_of_free_and_open-source_software_packages) software and [browsers](https://en.wikipedia.org/wiki/List_of_web_browsers). Adding a sub-section in the Firefox page under [3rd party forks](https://en.wikipedia.org/wiki/Firefox#Third-party_forks), or a new top level section, called "Ports to Windows OSes with dropped support" or something would help. These would probably help article writers on various topics too. 

Creating a Wikipedia page would give a lot of visibility/SEO. But it requires notability - an article at a popular site. Perhaps ask or write at [opensource.com](https://opensource.com/faq) ? , or a Windows 7 site to cover r3dfox... Maybe with some background on Microsoft's attempts to sideline Windows e.g. points mentioned above. Perhaps even add a page about this in the r3dfox website, as well as resources on useful tools like VxKek. Lists on Wikipedia don't require an article, they have far less notability. If a newbie removes the list entry citing notability just undo it and point that out, at worst it will need a news entry somewhere.

It's possible to send some info to be used as a news item to various tech sites like Techradar, BetaNews etc.

The other sites for probably announcing r3dfox would be, the various [Windows 7 forums](https://www.google.com/search?q="windows+7"+forums), [subreddit](https://old.reddit.com/r/windows7/comments/id5kbo/windows_7_survival_guide/)'s sticky thread needs updating for 2024, and other Windows sites.

**Comment by the-r3dacted** [comment]
> Once it's considered stable, probably sites like [Alternativeto.net](https://alternativeto.net/browse/new-apps/) . The big [download sites](https://www.google.com/search?q=software+download) that rank high on google like CNet or MajorGeeks, will accept download links from sourceforge as it's a major download site,

I've kinda seen those as somewhat shady, but yeah probably when I consider this to be release quality. I would say it's stable, but I wouldn't say it's release quality with the Aero issue, Firefox branding remnants, not really any privacy tweaks and such.

> But you probably should wait until you have a skeleton website for r3dfox as they like to link to a homepage as well.

Hmm, I guess I kinda already have that, but I should make it its own page. Also need to fix the typo. https://eclipse.cx/r3d.htm

> One thing you could do is title the sourceforge project something like "r3dfox - Modern Firefox in Windows 7"

Done.

I will look into posting it on GitLab and posting about it elsewhere. However for now I want to do some issue fixing.

**Comment by rovickti** [comment]
> Added to SourceForge. https://sourceforge.net/projects/r3dfox/

The big green download button https://sourceforge.net/projects/r3dfox/ and https://sourceforge.net/projects/r3dfox/files/ links to a 1GB source zip file instead of the latest installer. Is the source zip needed since the repo is available under the code section?

For later down the line:

> I've kinda seen those as somewhat shady, but yeah probably when I consider this to be release quality. 

Those used to to be major internet sites, back in the era of freeware, demos, shareware etc. software. They still have millions of downloads for things like Blender or GIMP. 

The issue with small opensource projects is getting the homepage noticed by google, in the era of search rankings, and saturation of tech news sites by marketing related to commercial products that send them info packages. The links from these sites, forums, reddit, articles, or Wikipedia help, Things like youtube videos don't get seen unless popular channels make a video. The marketing and ad supported internet makes non-profit things disappear into a blackhole easily.

Probably Makeuseof , [FSF directory](https://directory.fsf.org/wiki/Main_Page) and similar will help. Asking Windows forums to make a Windows 7/8 Survival guide sticky with all the useful software should get some google notice-ability and views. 

>Hmm, I guess I kinda already have that, but I should make it its own page. https://eclipse.cx/r3d.htm

You should probably add the new [homepage](https://sourceforge.net/p/forge/documentation/Create%20a%20New%20Project/#to-configure-your-project) on the sourceforge project as well, and link back to sourceforge downloads for SEO. Having a longer description on the sourceforge project with the words web browser should help search results too. Eventually having maybe a side-by-side screenshot of Firefox and r3dfox on the same page should help, and a youtube video. These would give something for news sites to copy.  Setting the categorisation tab in the [screenshot](https://sourceforge.net/p/forge/documentation/Create%20a%20New%20Project/) should help visibility in sourceforge.

There's a way to create custom external links to e.g. your donation url similar to what [hwinfo](https://sourceforge.net/projects/hwinfo/) did. It's possible to also make an external "Forum" link in a similar way, to the eclipse forum as well. 

I think sourceforge lets you create webpages with sourceforge.io (as well as your own domain), but the benefit of that subdomain inheriting search rankings is probably limited by google, since lots of small projects use it.

**Comment by the-r3dacted** [comment]
> The big green download button https://sourceforge.net/projects/r3dfox/ and https://sourceforge.net/projects/r3dfox/files/ links to a 1GB source zip file instead of the latest installer. Is the source zip needed since the repo is available under the code section?

Not sure why it defaulted to source, but fixed now.

For the other stuff, I will try to remember to do it later.

**Comment by the-r3dacted** [comment]
Since I finally am able to consider r3dfox release quality, I have posted it on other forums now.

https://msfn.org/board/topic/186106-r3dfox-modern-firefox-for-windows-7/

https://www.sevenforums.com/browsers-mail/429189-r3dfox-modern-firefox-windows-7-a.html

https://www.w7forums.com/threads/r3dfox-modern-firefox-for-windows-7.21939/ (Awaiting approval)

I'll also try Win-Raid, Bleeping Computer, and AskWoody later.

GitLab mirror, Reddit stuffs, Wikipedia stuffs coming later too.

**Comment by rovickti** [comment]
> I finally am able to consider r3dfox release quality

Congrats! :)

One thing I forgot to mention was the SEO boost, when you have finalised a domain/url for the homepage, if you ask r3dfox to be included at [portableapps](https://portableapps.com/about/contact) , [ninite](https://ninite.com/feedback) (they are picky, but since r3dfox is the only native Windows 7 Firefox there is no reason for them not to include it). And maybe even (eventually) release at chocolatey community repo using an [auto-updater](https://docs.chocolatey.org/en-us/create/create-packages).

Currently help ->about doesn't do an update check like it used to. Later on, if you are planning to have an update notification in the browser itself that has reliable uptime and lognevity, I guess it's possible to do a check of a special page under a sourceforge domain (and then give a link to open up the home/downloads page, or something?). That is, if you don't have a better option planned.

**Comment by the-r3dacted** [comment]
https://winraid.level1techs.com/t/r3dfox-modern-firefox-for-windows-7/100960

https://www.bleepingcomputer.com/forums/t/796349/r3dfox-modern-firefox-for-windows-7/

https://www.askwoody.com/forums/topic/r3dfox-modern-firefox-for-windows-7/

https://portableapps.com/node/70688

Also contacted Ninite, waiting to hear back.

I want to work on webpage now, then I'll likely move onto 125 RC.

I'll need to ask around and see what I can do for update server. Other forks have a updater, so I assume if they can do it, I can too.

**Comment by the-r3dacted** [comment]
GitLab mirror done https://gitlab.com/K4sum1/r3dfox

**Comment by rovickti** [comment]
Nice

To check how well the SEO from these improves visibility, does github give any download stats? What about clicks on the download link in the homepage?

Github is where most people will be downloading from since it's the first link mentioned. Sourceforge gives analytics by [file/folder](https://sourceforge.net/projects/r3dfox/files/124.0.3-hotfix/), including the OS the browser reports. 

Sourceforge: 51 downloads this week, 31 of those for latest release. 

Maybe categorizing the project in sourceforge might help increase visibility within sourceforge.

As for the number of Windows 7 users, and Firefox's share among those users, it's a bit more tricky.

Microsoft reported around July 2022 that there were [1.4 billion](https://www.techadvisor.com/article/745681/windows-now-has-1-4-billion-users-but-how-many-are-on-windows-11.html) Windows 10/11 PCs. Can assume most switched to different windows versions, as there were only a small increase in Linux after browsers discontinued support in early 2023. 

Statcounter results for Windows desktop OS distribution: 

[July 2022](https://gs.statcounter.com/windows-version-market-share/desktop/worldwide/#monthly-202206-202308): 10 (72.22%) + 11 (11.77%) = 83.99%. 

So 1.4 billion/0.8399 = 1.66 billion total windows PCs.

Assuming people who switched to Linux was small, and number of PCs remained mostly the same:

[Jan-April 2024 stats](https://gs.statcounter.com/windows-version-market-share/desktop/worldwide/#monthly-202401-202403): Windows 7 (3%).

3% of 1.66 billion = 49.8 million PCs give or take. That's still a lot of people.

Firefox had [7.21%](https://gs.statcounter.com/browser-market-share/desktop/worldwide/#monthly-202212-202302) of the browser share. Maybe the share increased on Windows 7 as Chrome didn't have an extended support period. But assuming all OSes had same proportions:

49.8*0.0721 = 3.59 million Firefox users, at least. 

So plenty of people who would really like to know about r3dfox, and maybe around 40 million Chrome based people who would be interested in knowing.

Firefox and Chrome dropped support in January 2023. You can see how Microsoft finally forced people to move by the drop starting in January, from [11% share](https://gs.statcounter.com/windows-version-market-share/desktop/worldwide/#monthly-202211-202403) and how the other two had an equivalent spike.

I'm not sure how to communicate to most of the 3.5 million people, in the era of marketing and SEO focused internet :(. I guess this is also why Microsoft gets so much advantage by bundling the Windows store, and bundling things like browsers.

Typing "Windows 7 browser" into google gives social media type results, and some guides like [this](https://mspoweruser.com/best-browser-windows-7/) or [this](https://windowsreport.com/best-browser-windows-7/) in browsers that work in Windows 7. Maybe contacting [here](https://windowsreport.com/contact-us/) or [here](https://mspoweruser.com/contact/) (or via "improve this article) option) might help. 

Maybe changing page/project titles to "modern Firefox for Windows 7" to "modern Firefox **browser** for Windows 7" would help with google searches.

**Comment by the-r3dacted** [comment]
> does github give any download stats? What about clicks on the download link in the homepage?

I don't think GitHub has anything. There is no tracking of any kind from my website, or nothing that I installed or know how to access.

> Maybe categorizing the project in sourceforge might help increase visibility within sourceforge.

Didn't even know that was a thing. Must have missed it when looking through settings. Categorized as a browser now.

> Typing "Windows 7 browser" into google gives social media type results, and some guides like [this](https://mspoweruser.com/best-browser-windows-7/) or [this](https://windowsreport.com/best-browser-windows-7/) in browsers that work in Windows 7. Maybe contacting [here](https://windowsreport.com/contact-us/) or [here](https://mspoweruser.com/contact/) (or via "improve this article) option) might help.

These look generic and/or shady click farm type content. Not sure if anyone trusts these.

> Maybe changing page/project titles to "modern Firefox for Windows 7" to "modern Firefox **browser** for Windows 7" would help with google searches.

Did for that for description on GitHub, SourceForge, and GitLab. Changed a few forum post titles where I was able to. Think it's good now.

**Comment by the-r3dacted** [comment]
So I finally got the webpage up, and I think it looks great.

https://eclipse.cx/projects/r3dfox.htm

**Comment by jonm58** [comment]
I put installer in china git:
https://gitee.com/CallCateIn58/r3dfox-installer

**Comment by the-r3dacted** [comment]
So tempted to slip b e a r into next release

**Comment by jonm58** [comment]
> So tempted to slip b e a r into next release

# Don't,lol

**Comment by rovickti** [comment]
> So I finally got the webpage up

Nice, might be worth adding the portable version in features list, and updating features/screenshots in sourceforge as well :)
(Does having exclamation marks after each feature make the project seem more trivial and sales like than the platforms only native Firefox (which is not a trivial thing), dunno.)

**Comment by the-r3dacted** [comment]
> Nice, might be worth adding the portable version in features list, and updating features/screenshots in sourceforge as well :)

Finally did it. I decided I would do it after I pushed 126, and I started working on new screenshots and adding features to the page afterwards. Also ran the screenshots on my site through cjpegli and made them smaller ~~even though I have 4TB of bandwith on my server why am I like this~~. Sourceforge is full quality png.

> (Does having exclamation marks after each feature make the project seem more trivial and sales like than the platforms only native Firefox (which is not a trivial thing), dunno.)

I don't know. I'm not a marketer. I wanted to make it sound right to me, and that made the most sense. If you have a better way to do it I would consider changing it.
